Packages/com.unity.inputsystem/InputSystem/Editor/PropertyDrawers/InputActionDrawerBase.cs

```diff
@@ -38,21 +50,56 @@ private void InitTreeIfNeeded(SerializedProperty property)
-                    onDoubleClick = OnItemDoubleClicked,
-                    title = (property.displayName, property.GetTooltip()),
-                    // With the tree in the inspector, the foldouts are drawn too far to the left. I don't
-                    // really know where this is coming from. This works around it by adding an arbitrary offset...
-                    foldoutOffset = 14,
-                    drawActionPropertiesButton = true
-                };
-                m_TreeView.Reload();
+                property.FindPropertyRelative(nameof(InputAction.m_Id)).stringValue = "";
+                property.FindPropertyRelative(nameof(InputAction.m_Name)).stringValue = "Input Action";
+                property.FindPropertyRelative(nameof(InputAction.m_SingletonActionBindings)).ClearArray();
+                property.serializedObject.ApplyModifiedProperties();
+            }
+                
+            viewData.TreeView = new InputActionTreeView(property.serializedObject)
+            {
+                onBuildTree = () => BuildTree(property),
+                onDoubleClick = item => OnItemDoubleClicked(item, property),
+                // With the tree in the inspector, the foldouts are drawn too far to the left. I don't
+                // really know where this is coming from. This works around it by adding an arbitrary offset...
+                foldoutOffset = 14,
+                drawActionPropertiesButton = true,
+                title = ("Input Action", property.GetTooltip())
+            };
+            viewData.TreeView.Reload();
+        }
+
+        private static bool IsPropertyAClone(SerializedProperty property)
+        {
+            // When a new item is added to a collection through the inspector, the default behaviour is
+            // to create a clone of the previous item. Here we look at all InputActions that appear before
+            // the current one and compare their Ids to determine if we have a clone. We don't look passed
+            // the current item because Unity will be calling this property drawer for each input action
+            // in the collection in turn. If the user just added a new input action, and it's a clone, as
+            // we work our way down the list, we'd end up thinking that an existing input action was a clone
+            // of the newly added one, instead of the other way around. If we do have a clone, we need to
+            // clear out some properties of the InputAction (id, name, and singleton action bindings) and
+            // recreate the tree view.
+
+            var array = property.GetArrayPropertyFromElement();
+            var index = property.GetIndexOfArrayElement();
+
+            for (var i = 0; i < index; i++)
+            {
+                if (property.FindPropertyRelative(nameof(InputAction.m_Id)).stringValue == array
+                    .GetArrayElementAtIndex(i).FindPropertyRelative(nameof(InputAction.m_Id)).stringValue) 
```

Comment:
Now occurs to me for the first time... did you check this with an `InputActionMap` property? `InputActionDrawerBase` is used as the basis both for `InputAction` and `InputActionMap` properties. Do the changes work for either?

Reply:
I did not...and they did not. But they do now.